lib/EnsEMBL/REST/Model/Regulatory.pm

```diff
@@ -237,6 +237,43 @@ sub get_microarray_info {
 
 }
 
+sub get_binding_matrix {
+    my ( $self, $binding_matrix_stable_id ) = @_;
+
+    my $c       = $self->context();
+    my $species = $c->stash->{species};
+    my $binding_matrix_adaptor =
+      $c->model('Registry')
+      ->get_adaptor( $species, 'Funcgen', 'BindingMatrix' );
+    my $binding_matrix =
+      $binding_matrix_adaptor->fetch_by_stable_id($binding_matrix_stable_id);
+
+    if ( !defined $binding_matrix ) {
+        Catalyst::Exception->throw( 'Binding Matrix '
+              . $binding_matrix_stable_id
+              . ' not found. Please check spelling.' );
+    }
+
+    my $unit = 'frequencies';
+    if ( defined $c->request->param('unit') ) {
+        $unit = $c->request->param('unit');
+    }
+
+    use Bio::EnsEMBL::Funcgen::BindingMatrix::Converter;
+    my $converter = Bio::EnsEMBL::Funcgen::BindingMatrix::Converter->new();
+
+    if ( $unit eq 'probabilities' ) {
+        $binding_matrix =
+          $converter->from_frequencies_to_probabilities($binding_matrix);
```

Comment:
This one is a little less safe...
https://github.com/Ensembl/ensembl-funcgen/blob/master/modules/Bio/EnsEMBL/Funcgen/BindingMatrix/Converter.pm#L106

Of course the Controller call to this is encapsulated in a try block. So it is caught before getting all the way back to Catalyst. But I'll give you this one, the user should probably get a more elegant message than "Assertion failed" in case of catastrophic failure. Is there a way to force a failure here in tests?